docs/autoprocess.py

```diff
@@ -0,0 +1,412 @@
+"""
+Sphinx extension for semi-autmoatic documentation of resolwe-bio processes.
+
+This module introduces new directive that enables user to document a
+set of processes. The directive can be used in three variations::
+
+    .. autoprocess::
+
+    .. autoprocess:: category_name
+
+    .. autoprocess:: process_slug
+
+The first option documents *all* processes.
+
+The second one documents just the ones that have *category* equal to
+`category_name`. Since some categories have subcategories (analyses,
+analyses:alignment, analyses:variants), this means that subcategories
+will *not* be documented.
+
+Third option decuments only one process: the one with *slug* equal to
+`process_slug`.
+"""
+
+import os
+import uuid
+import fnmatch
+
+import six
+import yaml
+
+from docutils import nodes
+from docutils.parsers.rst import Directive
+
+
+PROCESS_DIR = os.path.normpath(os.path.dirname(__file__) + "./../resolwe_bio/processes/")
+GITHUB_BASE_URL = 'https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/'
+
+
+class AutoProcessDirective(Directive):
+    """
+    Directive used to automatically document resolwe-bio process.
+    """
+    has_content = True
+    required_arguments = 0
+    optional_arguments = 0
+    final_argument_whitespace = False
+    option_spec = None
+
+    def filter_processes(self, content, categories, processes_dict):
+        """
+        Filter processes list by the content of the directive
+
+        Content can be:
+
+            * None - if there is not content, all processes are returned.
+            * Category - if the content of the directive is a string matching
+              an element in categories list, only processes with this
+              category will be returned.
+            * Process slug - if the content of directive is a valid process
+              slug only corresponding process is returned. Actually, a list
+              with only this process is returned.
+            * If none of this options, then raise a ValueError
+
+        :param content: content of the directive
+        :type content: list
+        :param categories: set of all categories
+        :type categories: set
+        :param processes: dict of all github_url: processes pairs
+        :type processes:dict
+
+        :rtype: list
+        """
+        if not content:
+            return processes
+        else:
```

Comment:
It will be more readable if you omit `else` and remove one level of indentation.